src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/sdl_pending_resumption_handler.h

```diff
@@ -0,0 +1,95 @@
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_RPC_PLUGINS_SDL_PLUGIN_INCLUDE_SDL_PLUGIN_SDL_PENDING_RESUMPTION_HANDLER_H
+#define SRC_COMPONENTS_APPLICATION_MANAGER_RPC_PLUGINS_SDL_PLUGIN_INCLUDE_SDL_PLUGIN_SDL_PENDING_RESUMPTION_HANDLER_H
+
+#include <map>
+#include <queue>
+#include <vector>
+#include "application_manager/event_engine/event_observer.h"
+#include "application_manager/resumption/extension_pending_resumption_handler.h"
+#include "application_manager/resumption/resumption_data_processor.h"
+#include "sdl_rpc_plugin/sdl_app_extension.h"
+
+namespace sdl_rpc_plugin {
+
+namespace app_mngr = application_manager;
+
+class SDLPendingResumptionHandler
+    : public resumption::ExtensionPendingResumptionHandler {
+ public:
+  SDLPendingResumptionHandler(
+      app_mngr::ApplicationManager& application_manager);
+
+  // EventObserver interface
+  void on_event(const app_mngr::event_engine::Event& event) OVERRIDE;
+
+  void HandleResumptionSubscriptionRequest(
+      app_mngr::AppExtension& extension,
+      resumption::Subscriber& subscriber,
+      application_manager::Application& app) OVERRIDE;
+
+  void OnResumptionRevert() OVERRIDE;
+
+ private:
+  /**
+   * @brief RaiseFakeSuccessfulResponse raize event for the subscriber that
+   * contains emulated successful response from HMI To avoid double subscription
+   * SDLPendingResumptionHandler freezes sending requests to HMI. But
+   * subscriber() function need to be called to provide information that some
+   * data need to be resumed. So if pending request exists, SDL creates
+   * preliminary requests to HMI, subscribe the subscriber to this request but
+   * do not send it to HMI. It freezes this requests to precess the one by one
+   * when a response to the current request will be received When SDL receives a
+   * response from HMI it may satisfy some frozen requests. If it does SDL will
+   * create faked HMI response(based on the real one with corr_id replacement)
+   * and raize event. So that subscriber::on_event will be called with an
+   * appropriate response to the request that SDL was not sent to HMI.
+   * @param response message that will be raised with corr_id replacement
+   * @param corr_id correlation id that will be replaced in response to notify
+   * the subscriber
+   */
+  void RaiseFakeSuccessfulResponse(smart_objects::SmartObject response,
+                                   const int32_t corr_id);
+  smart_objects::SmartObjectSPtr CreateSubscriptionRequest();
+
+  struct ResumptionAwaitingHandling {
+    const uint32_t app_id;
+    SDLAppExtension& ext;
+    resumption::Subscriber subscriber;
+    resumption::ResumptionRequest request_to_send_;
+  };
+
+  typedef std::pair<SDLAppExtension, resumption::Subscriber> FreezedResumption;
+  std::vector<ResumptionAwaitingHandling> freezed_resumptions_;
```

Comment:
```suggestion
  std::vector<ResumptionAwaitingHandling> frozen_resumptions_;
```

Reply:
@iCollin renamed in https://github.com/smartdevicelink/sdl_core/pull/3475/commits/4949821a9f4a08f540af1443fc92d2060c296159